### PR TITLE
Transpile @vector-im/compound-web with babel

### PR DESCRIPTION
**Pull request by germain-gg**
Fixes https://github.com/vector-im/element-web/issues/25905
For https://github.com/vector-im/element-web/issues/25883

To review alongside https://github.com/matrix-org/matrix-react-sdk/pull/11354

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/vector-im/element-web/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by t3chguy** [comment]
@germain-gg could you include the rationale for this in the PR description? Why not just consume the built artifacts from the library like we do for the vast majority of our dependencies (both internal & external)

**Comment by germain-gg** [comment]
@t3chguy we essentially need it because it otherwise throws

```
2023-08-02 10:29:36.683 [element-js] ERROR in ../matrix-react-sdk/node_modules/@vector-im/compound-web/dist/compound-web.js 1482:22
2023-08-02 10:29:36.683 [element-js] Module parse failed: Unexpected token (1482:22)
2023-08-02 10:29:36.683 [element-js] You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
2023-08-02 10:29:36.683 [element-js] |         onMouseDown: (e) => {
2023-08-02 10:29:36.683 [element-js] |           e.preventDefault();
2023-08-02 10:29:36.683 [element-js] >           onMouseDown?.(e);
2023-08-02 10:29:36.683 [element-js] |         }
2023-08-02 10:29:36.683 [element-js] |       },
```

We still use the `?.` optional chaining operator in the original bundle because our browser support does not require us to transpile it, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility 

But our webpack/babel setup is tricking us in having to do it anyway. And i'm not looking to fix this at this stage.

File: webpack.config.js [hunk 1]
```diff
@@ -271,6 +271,12 @@ module.exports = (env, argv) => {
                         if (f.startsWith(reactSdkSrcDir)) return true;
                         if (f.startsWith(jsSdkSrcDir)) return true;
 
+                        // Some of the syntax in this package is not understood by
+                        // either webpack or our babel setup.
+                        // When we do get to upgrade our current setup, this should
+                        // probably be removed.
+                        if (f.includes("compound-web")) return true;
```
Comment (t3chguy) [inline webpack.config.js:278]: Could we be more specific, there might be transitive imports which also contain the text `compound-web`

**Review by t3chguy — CHANGES_REQUESTED** [review]

**Review by t3chguy — APPROVED** [review]
LGTM

**Pull request merged**